internal/input/trades.go

```diff
@@ -0,0 +1,97 @@
+package input
+
+import (
+	"time"
+
+	ingestio "github.com/stellar/go/exp/ingest/io"
+	"github.com/stellar/go/xdr"
+	"github.com/stellar/stellar-etl/internal/utils"
+)
+
+// TradeTransformInput is a representation of the input for the TransformTrade function
+type TradeTransformInput struct {
+	OperationIndex int32
+	Transaction    ingestio.LedgerTransaction
+	CloseTime      time.Time
+}
+
+// GetTrades returns a slice of trades for the ledgers in the provided range (inclusive on both ends)
+func GetTrades(start, end uint32, limit int64) ([]TradeTransformInput, error) {
+	backend, err := utils.CreateBackend()
+	if err != nil {
+		return []TradeTransformInput{}, err
+	}
+
+	defer backend.Close()
+
+	latestNum, err := backend.GetLatestLedgerSequence()
+	if err != nil {
+		return []TradeTransformInput{}, err
+	}
+
+	err = validateLedgerRange(start, end, latestNum)
+	if err != nil {
+		return []TradeTransformInput{}, err
+	}
+
+	tradeSlice := []TradeTransformInput{}
+	for seq := start; seq <= end; seq++ {
+		txReader, err := ingestio.NewLedgerTransactionReader(backend, publicPassword, seq)
+		if err != nil {
+			return []TradeTransformInput{}, err
+		}
+
+		closeTime, err := utils.TimePointToUTCTimeStamp(txReader.GetHeader().Header.ScpValue.CloseTime)
+		if err != nil {
+			return []TradeTransformInput{}, err
+		}
+
+		for int64(len(tradeSlice)) < limit || limit < 0 {
+			tx, err := txReader.Read()
+			if err == ingestio.EOF {
+				break
+			}
+
+			for index, op := range tx.Envelope.Operations() {
+				// Trades occur on these operation types: manage buy offer, manage sell offer,
+				// create passive sell offer, path payment send, and path payment receive
+				// Trades also can only occur when these operations are successful
+				if operationResultsInTrade(op) && tx.Result.Successful() {
```

Comment:
These are necessary conditions, but are these sufficient for an operation to be a trade? For example, wouldn't this mean a successful offer operation would count as a trade?

Reply:
Not all of these operations will result in trades, but this is checked in TransformTrade (an empty slice is returned if no trades occurred) for a given op.